treeshr/RemoteAccess.c

```diff
@@ -147,9 +147,9 @@ static int remote_access_disconnect(int conid, int force){
     DBG("Connection %d< %d\n",conid,host->h.connections);
   } else DBG("Disconnected %d\n",conid);
   for (host = host_list; host ;) {
-    if ((force && host->h.conid == conid) || (host->h.connections <= 0
+    if ((force && host->h.conid == conid) || (host->h.conid == conid && host->h.connections <= 0
```

Comment:
the second 'host->h.conid == conid' would eventually leave old connections alive when USE_TIME is true. this second loop seems to only make sense with USE_TIME as it all happens in a HOST_LIST lock anyway.
i will clean it up.

Reply:
In any case the previous version was wrong because if any element is found with h.connections == 0, then DisconnectFromMds() was called ***with the wrong conid*** that is the conid passed to remote_access_disconnect().

This was the reason for the error below that drove us crazy for months

Error in SendArg: mode = 1, status = 65554
Error in SendArg: mode = 6, status = 65554
Error in SendArg: mode = 6, status = 65554
Error writing device data: DatabaseException: %TREE-E-FAILURE, Operation NOT successful
null